src/gsSolver/gsBiCgStab.hpp

```diff
@@ -0,0 +1,90 @@
+/** @file gsBiCgStab.hpp
+
+    @brief Biconjugate gradient stabilized solver
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): R. Tielen, S. Takacs
+*/
+
+namespace gismo
+{
+
+template<class T>
+bool gsBiCgStab<T>::initIteration( const typename gsBiCgStab<T>::VectorType& rhs,
+                                         typename gsBiCgStab<T>::VectorType& x )
+{
+    if (Base::initIteration(rhs,x))
+        return true;
+
+    // m_res = rhs - A * x;
+    m_mat->apply(x,m_tmp);
+    m_res = rhs - m_tmp;
+
+    m_r0 = m_res;
+
+    m_p.resize( m_mat->cols(), 1); m_p.setZero();
+    m_v.resize( m_mat->cols(), 1); m_v.setZero();
```

Comment:
Could be shortened as 
m_p.setZero( m_mat->cols(), 1);
m_v.setZero( m_mat->cols(), 1);